frontend/src/pages/detail/ModfiyBook.tsx

```diff
@@ -105,169 +111,205 @@ export default function BookDetail() {
         height: height,
         thickness: thickness,
         stock: 0,
+        urls: [""],
       };
       logger.debug("Submitting Book Modify", book);
       BOOKS_API.modifyBook({ book: book })
-        .then(() => showSuccess())
-        .catch(() => showFailure("Could not modify book"));
+        .then(() => showSuccess(toast, "Book Edited"))
+        .catch(() => showFailure(toast, "Could not modify book"));
       formik.resetForm();
     },
   });
 
+  useEffect(() => {
+    IMAGES_API.getImage({ id: id })
+      .then((response) =>
+        setImage({
+          imageSrc: response.url,
+          imageHash: Date.now(),
+        })
+      )
+      .catch((error) =>
+        showFailure(toast, "Image Cannot be Retrieved to Update Display")
+      );
+  }, []);
+
+  //Restore the original Page load data back when edit toggle is turned off
+  useEffect(() => {
+    if (!isModifiable) {
+      setGenre(originalBookData.genres);
+      setHeight(originalBookData.height);
+      setWidth(originalBookData.width);
+      setThickness(originalBookData.thickness);
+      setPageCount(originalBookData.pageCount);
+      setPrice(originalBookData.retailPrice);
+    }
+  }, [isModifiable]);
+
+  // The dropdown configuration for each cell
   const genreDropdown = GenreDropdown({
     setSelectedGenre: setGenre,
     selectedGenre: genre,
+    isModifiable: !isModifiable,
   });
 
+  const uploadImageFileHandler = (event: FileUploadHandlerEvent) => {
+    const file = event.files[0];
+    IMAGES_API.uploadImage({ id: id, image: file })
+      .then((response) => {
+        IMAGES_API.getImage({ id: id })
+          .then((response) =>
```

Comment:
Can you remove the getImage after the upload as we talked about in class

Reply:
I have went back and tested it the reason I think we should leave it in is because the url hash needs to update, and that occurs here. Also in order for the url for the image to be updated as a React state change in the Image component it needs a get request that causes the browser to ignore the cache. Not sure why, but currently it works like this, will look more into it in the future.

Reply:
Makes sense, if we need more images in the future we can look into an alternative